crates/core/tedge_api/src/workflow/mod.rs

```diff
@@ -249,8 +268,8 @@ impl OperationWorkflow {
 }
 
 impl OperationAction {
-    pub fn with_default(self, default: &DefaultHandlers) -> Self {
-        match self {
+    pub fn with_default(self, default: &DefaultHandlers) -> Result<Self, WorkflowDefinitionError> {
```

Comment:
I would not return errors here. We need to distinguish "definition consistency checking" from "default values insertion". 

```suggestion
    pub fn with_default(self, default: &DefaultHandlers) -> Self
```


I think there are sensible defaults in most cases. 
For instance, the default "on_success" step for the iterate action can simply be the "successful" state.

That said, I agree that the way default values are managed can be improved. There are too many places where the workflow actor has to forge default steps. => I will address that in a follow-up PR.

Reply:
I was under the impression that the default values can be used only when default handlers are explicitly defined in the workflow. An implicit default of `failed` for an `on_error` target might be fine, but is it okay to implicitly default to the `successful` state when it is not specified explicitly? The iterate action is most likely an intermediate state in a workflow, and even after the iteration completes, there could be additional states that the workflow needs to go through, before deeming the execution as successful. So, if we implicitly default to the successful state right after the iteration (when one was not specified accidentally), critical steps in the workflow might get skipped. So, isn't it better to force the user to be explicit with the `on_success` target?

Reply:
> So, isn't it better to force the user to be explicit with the `on_success` target?

This  can be discussed. From my point of view, the `successful` state makes sense as a default `on_success` target.

However, the main point of my comment is not here. Rather than on the UX, I was pointing out code organization.
The `with_default` method should only inject default values and not check correctness.

Reply:
Resolved by 1012f41c0069b302fb54aa249149fdee9ab20071